ibis/expr/streaming.py

```diff
@@ -0,0 +1,9 @@
+from __future__ import annotations
+
+import ibis.expr.types as ir  # noqa: TCH001
+from ibis.common.grounds import Concrete
+
+
+class Watermark(Concrete):
+    time_col: str  # TODO(chloeh13q): allow column references
```

Comment:
Since these will always be used as part of a `create_table` call, I don't see how the user would ever have access to a direct `Column` here. Supporting only a string name seems fine to me. I'd drop this TODO entirely.